src/Model/State.re

```diff
@@ -80,13 +79,34 @@ type t = {
 };
 
 let initial = (~getUserSettings, ~contributedCommands, ~workingDirectory) => {
-  let editorGroups = EditorGroups.create();
-  let initialEditorGroup = editorGroups |> EditorGroups.getFirstEditorGroup;
+  Vim.init();
```

Comment:
This is not mean to be called multiple times (and there's no protection against it, currently).

Reply:
Is `State.initial` called multiple times? It's not supposed to be, although purity seems like a reasonable assumption for this, so I can easily see it happening in the future.

Reply:
> Is State.initial called multiple times?

I don't think it is, today

>  It's not supposed to be, although purity seems like a reasonable assumption for this, so I can easily see it happening in the future.

Ya, exactly, this is my main concern - it could be easy to break something unexpectedly

Reply:
Fixed in https://github.com/onivim/oni2/pull/1977/commits/b19acca26fef78f0c80cc633d1aa15b3be419cde